engine/collection/epochmgr/engine.go

```diff
@@ -1,233 +1,189 @@
+	epochStopEvents              chan uint64       // sends counter of epoch to stop
+
+	cm *component.ComponentManager
+	component.Component
 }
 
+var _ component.Component = (*Engine)(nil)
+
 func New(
 	log zerolog.Logger,
 	me module.Local,
 	state protocol.State,
-	pools *epochs.TransactionPools,
+	pools *epochpool.TransactionPools,
 	voter module.ClusterRootQCVoter,
 	factory EpochComponentsFactory,
 	heightEvents events.Heights,
 ) (*Engine, error) {
-	ctx, stopComponents := context.WithCancel(context.Background())
-	signalerCtx, _ := irrecoverable.WithSignaler(ctx)
 
 	e := &Engine{
-		unit:             engine.NewUnit(),
-		log:              log.With().Str("engine", "epochmgr").Logger(),
-		me:               me,
-		state:            state,
-		pools:            pools,
-		voter:            voter,
-		factory:          factory,
-		heightEvents:     heightEvents,
-		epochs:           make(map[uint64]*StartableEpochComponents),
-		startupTimeout:   DefaultStartupTimeout,
-		irrecoverableCtx: signalerCtx,
-		stopComponents:   stopComponents,
+		log:                          log.With().Str("engine", "epochmgr").Logger(),
+		me:                           me,
+		state:                        state,
+		pools:                        pools,
+		voter:                        voter,
+		factory:                      factory,
+		heightEvents:                 heightEvents,
+		epochs:                       make(map[uint64]*StartableEpochComponents),
+		startupTimeout:               DefaultStartupTimeout,
+		epochTransitionEvents:        make(chan *flow.Header, 1),
+		epochSetupPhaseStartedEvents: make(chan *flow.Header, 1),
+		epochStopEvents:              make(chan uint64, 1),
 	}
 
+	e.cm = component.NewComponentManagerBuilder().
+		AddWorker(e.handleEpochSetupPhaseStartedLoop).
+		AddWorker(e.handleEpochTransitionLoop).
+		AddWorker(e.handleStopEpochLoop).
+		Build()
+	e.Component = e.cm
+
+	return e, nil
+}
+
+// Start starts the engine.
+func (e *Engine) Start(ctx irrecoverable.SignalerContext) {
+	// 1 - start engine-scoped workers
+	e.cm.Start(ctx)
+
+	// 2 - check if we should attempt to vote after startup
+	err := e.checkShouldVoteOnStartup()
+	if err != nil {
+		ctx.Throw(err)
+	}
+
+	// 3 - start epoch-scoped components
 	// set up epoch-scoped epoch managed by this engine for the current epoch
 	epoch := e.state.Final().Epochs().Current()
 	counter, err := epoch.Counter()
 	if err != nil {
-		return nil, fmt.Errorf("could not get epoch counter: %w", err)
+		ctx.Throw(err)
 	}
-
 	components, err := e.createEpochComponents(epoch)
-	// don't set up consensus components if we aren't authorized in current epoch
-	if errors.Is(err, ErrNotAuthorizedForEpoch) {
-		return e, nil
+	if err != nil {
+		if errors.Is(err, ErrNotAuthorizedForEpoch) {
+			// don't set up consensus components if we aren't authorized in current epoch
+			e.log.Info().Msg("node is not authorized for current epoch - skipping initializing cluster consensus")
+			return
```

Comment:
Yes we do, because at the next epoch transition we will need to handle the protocol event and start up consensus for the new epoch. We also need to respond to events in order to submit our root block vote, even if we aren't authorized for the current epoch.